### PR TITLE
Bump to SBT 1.9, bump libraries, remove deprecations.

### PR DESCRIPTION
**Pull request by Nezisi**
- bump to 0.12.0-SNAPSHOT
- migrate SBT to 1.9.0, drop support for SBT 0.13 / 1.0.0
- bump JUnit / JUnit 4 / Hamcrest / Mockito
- bump Scala to 2.12.18, remove cross scala compiling
- set minimum JDK version to 8, thus source / target to 1.8
- add encoding / xlint settings 

Compilation was tested with JDK 8, JDK 11, JDK 17. 

I didn't test whether the release plugin / sonatype plugin still works, as I'm not familiar with it.

### CODE REVIEW

File: src/library/src/main/java/net/aichler/jupiter/internal/event/LoggingEventHandler.java [hunk 1]
```diff
@@ -32,7 +32,7 @@
 /**
  * Writes dispatch events to a file for testing/debugging purposes.
  *
- * <h3>Example Usage</h3>
+ * <h2>Example Usage</h2>
```
Comment (Nezisi) [inline src/library/src/main/java/net/aichler/jupiter/internal/event/LoggingEventHandler.java:35]: This change was necessary as the doc compiler on JDK 17 generated an error: 
heading used out of sequence: <H3>, compared to implicit preceding heading: <H1>

**Comment by mkurz** [comment]
You could use sbt 1.9.0 already.

**Comment by Nezisi** [comment]
@mkurz Jesus, that was fast. :) Yes, bumped to SBT 1.9.

**Comment by mkurz** [comment]
:wink: Title of the PR still says sbt 1.8...

**Comment by mkurz** [comment]
@maichler do you need help maintaining this library? I have lots of experience with sbt-plugins, maintaining dozens of them within the sbt organization. Would you mind add me as maintainer so I can take a look at pull requests and merge them?
What do you think?

**Comment by maichler** [comment]
> @maichler do you need help maintaining this library? 

Hi @mkurz :wave: 
I haven't been able to find the time for this project for a while now :( So yeah, I guess I could really use some help there.

> I have lots of experience with sbt-plugins, maintaining dozens of them within the sbt organization. Would you mind add me as maintainer so I can take a look at pull requests and merge them? What do you think?

I think you're just the person this project needs right now, thanks a lot for the offer!

Let's give it a shot.

**Comment by mkurz** [comment]
@maichler Also, if you want to keep the groupId `net.aichler`, we still could set up release by tag via https://github.com/sbt/sbt-ci-release. You would need to add some secrets yourself though: https://github.com/sbt/sbt-ci-release?tab=readme-ov-file#secrets

If you decide to change the groupId, you could make me admin of this repo and I can add credentials.

**Comment by maichler** [comment]
> @maichler Perfect, thanks! Also what do you think about changing the groupId from `net.aichler` to `com.github.sbt`? 

I don't mind changing the groupId, the only reason I kept it was to not break bc.

> Because this repos already lives under the GitHub sbt organization, and not just that, we could set up an automatic release process via GitHub actions so with just pushing a tag a release gets published to maven central automatically. I did set up such workflows in many sbt projects already with that groupId.

That would simplify the process a lot, I like it :+1:

**Comment by mkurz** [comment]
Great, thanks! I will come back to set up things this week.

**Comment by mkurz** [comment]
@maichler Just to let you know, I renamed the `master` branch to `main`, because this is more common now and all repos I work with a named that way (also actually the reusable workflows I will introduce expecte the default branch to be named ´main`).

**Comment by sleski** [comment]
Hallo @mkurz, any info regarding merging this PR? Thanks

**Comment by mkurz** [comment]
@sleski I plan to work on this in July/August and to integrate it into Play Framework.

**Comment by eed3si9n** [comment]
👋🏼 I wanted the PR to split the library bump and sbt bump so I sent https://github.com/sbt/sbt-jupiter-interface/pull/75 for the sbt + GitHub Actions bumps.

**Comment by eed3si9n** [comment]
Scala Steward took care of the JUnit 5 bump, and I've cherry picked the hamcrest bump to https://github.com/sbt/sbt-jupiter-interface/pull/83 + CI checking on Zulu JDK 8 and Temurin JDK 17.

**Pull request closed without merging**